lib/posix/options/pthread.c

```diff
@@ -1114,6 +1115,41 @@ int pthread_join(pthread_t pthread, void **status)
 	return 0;
 }
 
+/**
+ * @brief Await a thread termination with timeout.
+ *
+ * Non-portable GNU extension of IEEE 1003.1
+ */
+int pthread_timedjoin_np(pthread_t pthread, void **status, const struct timespec *abstime)
+{
+	if (abstime->tv_sec < 0 || abstime->tv_nsec < 0 || abstime->tv_nsec >= NSEC_PER_SEC) {
```

Comment:
This should have test coverage as well
 ```suggestion
	if (abstime == NULL) {
		return EINVAL;
	}
	if (abstime->tv_sec < 0 || abstime->tv_nsec < 0 || abstime->tv_nsec >= NSEC_PER_SEC) {
```